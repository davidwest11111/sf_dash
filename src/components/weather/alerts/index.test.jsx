import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import WeatherAlerts from './index';


const SAMPLE_ALERT = {
  title: 'Flood Watch for Mason, WA',
  time: 1509993360,
  expires: 1510036680,
  description: '...FLOOD WATCH REMAINS IN EFFECT THROUGH LATE MONDAY NIGHT...\n',
  uri: 'http://alerts.weather.gov/cap/wwacapget.php?x=WA1255E4DB8494.FloodWatch.1255',
};

describe('outputs the expected tree when', () => {
  let wrapper;

  test('(default)', () => {
    wrapper = shallow((
      <WeatherAlerts />
    ));
  });

  test('there are alerts', () => {
    wrapper = shallow((
      <WeatherAlerts alerts={[SAMPLE_ALERT]} />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
