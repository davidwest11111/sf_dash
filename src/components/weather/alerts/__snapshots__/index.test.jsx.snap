// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `""`;

exports[`outputs the expected tree when there are alerts 1`] = `
<div
  className="col-12 alerts"
>
  <ul>
    <li
      key="Flood Watch for Mason, WA"
    >
      Flood Watch for Mason, WA
    </li>
  </ul>
</div>
`;
